google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/FirestoreTest.java

```diff
@@ -169,4 +181,86 @@ public void incrementEquals() {
     assertNotEquals(increment1, increment3);
     assertNotEquals(increment2, increment4);
   }
+
+  @Test
+  public void arrayUnionWithPojo() throws ExecutionException, InterruptedException {
+    doReturn(commitResponse(1, 0))
+        .when(firestoreMock)
+        .sendRequest(
+            commitCapture.capture(), Matchers.<UnaryCallable<CommitRequest, CommitResponse>>any());
+    DocumentReference doc = firestoreMock.document("coll/doc1");
+    FieldValue fieldValue =
+        FieldValue.arrayUnion(
+            new TestPoJo(
+                1,
+                "name,",
+                ImmutableList.<List<String>>of(ImmutableList.<String>of("test-1", "test-2")),
+                ImmutableMap.<String, Map<String, String>>of(
+                    "test-1", ImmutableMap.of("test-value1", "test-value2")),
+                ImmutableList.<Map<String, String>>of(
+                    ImmutableMap.<String, String>of("test", "test-value"))));
+    WriteResult writeResult = doc.update("foo", fieldValue).get();
+    assertEquals(Timestamp.ofTimeSecondsAndNanos(0, 0), writeResult.getUpdateTime());
```

Comment:
I know it would be a lot more code, but to me this test would be more valuable if it verified the generated proto.